libs/features/mas/commerce/src/mas-commerce-service.js

```diff
@@ -0,0 +1,183 @@
+     * @param config, if not provided, the one from attributes of the element will be taken
+     * @param providers, if not provided, the one from registrations before activation will be taken
+     * @returns 
+     */
+    async innerActivate(resolve, config, dataProviders) {
+        // Load settings and literals
+      const log = Log.init(config.env).module('service');
+      log.debug('Activating:', config);
+      const literals = { price: {} };
+      const settings = Object.freeze(getSettings(config));
+      // Fetch price literals
+      try {
+        literals.price = await getPriceLiterals(settings, config.commerce.priceLiterals);
+      } catch (error) {
+        log.warn('Price literals were not fetched:', error);
+      }
+      // Create checkout/price options providers registry
+      const providers = { 
+        /** @type {Set<Commerce.Checkout.provideCheckoutOptions>} */
+        checkout: new Set(),
+        /** @type {Set<Commerce.Price.providePriceOptions>} */
+        price: new Set(),
+      };
+      this.startup = { literals, providers, settings };      
+      // Extend web component object with service API
+      // @ts-ignore
+      MasCommerceService.instance = Object.defineProperties(
+        this,
+        Object.getOwnPropertyDescriptors({
+          // Activate modules and expose their API as combined flat object
+          ...Checkout(this.startup, dataProviders),
+          ...Ims(this.startup),
+          ...Price(this.startup),
+          ...Wcs(this.startup),
+          ...Constants,
+          // Defined serviceweb  component API
+          Log,
+          get defaults() {
+              return Defaults;
+          },
+          get literals() {
+              return literals;
+          },
+          get log() {
+              return Log;
+          },
+          get providers() {
+              return {
+                  checkout(provider) {
+                      providers.checkout.add(provider);
+                      return () => providers.checkout.delete(provider);
+                  },
+                  price(provider) {
+                      providers.price.add(provider);
+                      return () => providers.price.delete(provider);
+                  },
+              };
+          },
+          get settings() {
+              return settings;
+          },
+        }),
+      );
+      log.debug('Activated:', { literals, settings, });
+      setImmediate(() => {
+        const event = new CustomEvent(EVENT_TYPE_READY, {
+          bubbles: true,
+          cancelable: false,
+          detail: MasCommerceService.instance,
+        });
+        this.dispatchEvent(event);
+      });
+      resolve(this);
+    }
+
+    /**
+     * @param config, if not provided, the one from attributes of the element will be taken
+     * @param providers, if not provided, the one from registrations before activation will be taken
+     * @returns
+     */
+    async activate(config, dataProviders) {
```

Comment:
I strongly think that we should remove active/deactivate/reset features.
Just thing as a custom element, where it is attached and detached from DOM.

Reply:
ok, we can discuss that, we have a consumer willing to activate things themselves (but they can adding the component to the DOM i guess :) )